### PR TITLE
Add a submit queue pipeline

### PR DESCRIPTION
**Pull request by 0xmichalis**
@dobbymoodge @leifmadsen you guys have expressed interest for this stuff.

@stevekuznetsov @jupierce

### CODE REVIEW

**Comment by stevekuznetsov** [comment]
I still strongly suggest not going the api aggregation route. We're not ready in Origin for this. The path of least resistance is to start up a Kubernetes cluster in GKE for now and let it host the infra. Only thing I can see making that difficult is if we need endpoints in that cluster to have nice stable names for webhook delivery from GitHub. 

/cc @smarterclayton

**Comment by kargakis** [comment]
I am fine with using GKE, prow was running fine there last time I checked. I was worried about the migration of ProwJobs from one cluster to another but eventually I don't think we really care about it.

**Comment by stevekuznetsov** [comment]
Ok. With the known unknowns, it looks like GKE will be the path of least resistance. Lets investigate the unknown unknowns. Can we host other components in the OpenShift cluster? Where are we hooks being delivered?
 Do we need a SSL cert so we can get web hook payloads over HTTPS into the GKE cluster? Can we terminate in the OpenShift cluster and forward to GKE? Does any part of the ecosystem today require other features we'll only find in GKE, like bleeding edge Kubernetes stuff we won't have in OpenShift? Will that impede us migrating when TPR become CRD?

**Comment by kargakis** [comment]
The submit queue can run on Openshift as it is not interacting with the cluster. Prow can run on GKE. HTTPS shouldn't be required for the poc and the payload can still be delivered in GKE last time I checked but once we are ok with switching, sq, hook, and deck will need to use HTTPS. Generally, prow shouldn't make assumptions for the platform it's running on and the remaining work for making it agnostic has to do with bot names, namespaces, and the kubernetes org is hardcoded in some places (nothing GKE-specific).

**Comment by kargakis** [comment]
Regarding the migration, I think it's a straightforward export of the ProwJobs from one cluster to another and we don't necessarily need to do it as their value is purely historical at this point.

**Comment by stevekuznetsov** [comment]
Wait, as long as we can deploy `hook` in OpenShift and only `prow` in GKE, we are fine on HTTPS webhooks.

**Comment by kargakis** [comment]
hook creates ProwJobs based on the events it gets from the webhooks.

**Comment by stevekuznetsov** [comment]
@kargakis but does `hook` itself need to be in the GKE cluster? How does it authenticate? Can it just send the API requests into the GKE cluster from outside?

**Comment by kargakis** [comment]
All of the components are using an [in-cluster client](https://github.com/kubernetes/test-infra/blob/2b8328fe6a5bde56b9dcd4423cc3c1257990681b/prow/kube/client.go#L196) that talks to https://kubernetes and [hook is no different](https://github.com/kubernetes/test-infra/blob/2b8328fe6a5bde56b9dcd4423cc3c1257990681b/prow/cmd/hook/main.go#L110). Why can't we have https webhooks in GKE?

**Comment by stevekuznetsov** [comment]
> Why can't we have https webhooks in GKE?

Don't we need certs to terminate SSL there? I'm not super familiar with how we need to set that up. My worry is that we will want there to be very little time between PoC and deployment here so if there is something slow we need to trigger now -- like requesting certs -- we should do it while we we work on PoC.

**Comment by kargakis** [comment]
[Ingress controller](https://github.com/kubernetes/ingress) and something like [lego](https://github.com/jetstack/kube-lego). Upstream seem to be using lego for minting certificates, too.

https://github.com/kubernetes/test-infra/blob/master/prow/cluster/lego_deployment.yaml

**Comment by stevekuznetsov** [comment]
SGTM

**Comment by kargakis** [comment]
@pospispa

File: cluster/ci/config/prow/proxy/deck_proxy.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+metadata:
+  name: deck-proxy
+spec:
+  template:
+    metadata:
+      labels:
+        app: deck
+    spec:
+      containers:
+      - image: gcr.io/google-containers/nginx:test-cmd
```
Comment (kargakis) [inline cluster/ci/config/prow/proxy/deck_proxy.yaml]: We likely want to use a different image that the current.

**Comment by kargakis** [comment]
Updated this PR to contain only a pipeline for the SubmitQueue

https://github.com/openshift/release/pull/44 adds prow manifests
https://github.com/openshift/release/pull/45 adds the GKE proxies

**Pull request merged**